Safe Management/Safe-Management.ps1

```diff
@@ -184,217 +216,263 @@ Function Write-LogMessage
-				$msgToWrite += "[WARNING]`t$Msg"
-			}
-			"Error" {
-				Write-Host $MSG.ToString() -ForegroundColor Red
-				$msgToWrite += "[ERROR]`t$Msg"
-			}
-			"Debug" { 
-				if($InDebug)
-				{
-					Write-Debug $MSG
-					$msgToWrite += "[DEBUG]`t$Msg"
-				}
-				else { $writeToFile = $False }
-			}
-			"Verbose" { 
-				if($InVerbose)
-				{
-					Write-Verbose $MSG
-					$msgToWrite += "[VERBOSE]`t$Msg"
-				}
-				else { $writeToFile = $False }
-			}
-		}
+        $msgToWrite = "[$(Get-Date -Format "yyyy-MM-dd hh:mm:ss")]`t"
+        $writeToFile = $true
+        # Replace empty message with 'N/A'
+        if ([string]::IsNullOrEmpty($Msg))
+        {
+            $Msg = "N/A" 
+        }
+        # Mask Passwords
+        if ($Msg -match '((?:"password"|"secret"|"NewCredentials")\s{0,}["\:=]{1,}\s{0,}["]{0,})(?=([\w!@#$%^&*()-\\\/]+))')
+        {
+            $Msg = $Msg.Replace($Matches[2], "****")
+        }
+        # Check the message type
+        switch ($type)
+        {
+            "Info"
+            { 
+                Write-Host $MSG.ToString()
+                $msgToWrite += "[INFO]`t$Msg"
+            }
+            "Warning"
+            {
+                Write-Host $MSG.ToString() -ForegroundColor DarkYellow
+                $msgToWrite += "[WARNING]`t$Msg"
+            }
+            "Error"
+            {
+                Write-Host $MSG.ToString() -ForegroundColor Red
+                $msgToWrite += "[ERROR]`t$Msg"
+            }
+            "Debug"
+            { 
+                if ($InDebug)
+                {
+                    Write-Debug $MSG
+                    $msgToWrite += "[DEBUG]`t$Msg"
+                }
+                else
+                {
+                    $writeToFile = $False 
+                }
+            }
+            "Verbose"
+            { 
+                if ($InVerbose)
+                {
+                    Write-Verbose $MSG
+                    $msgToWrite += "[VERBOSE]`t$Msg"
+                }
+                else
+                {
+                    $writeToFile = $False 
+                }
+            }
+        }
 		
-		If($writeToFile) { $msgToWrite | Out-File -Append -FilePath $LOG_FILE_PATH }
-		If ($Footer) { 
-			"=======================================" | Out-File -Append -FilePath $LOG_FILE_PATH 
-			Write-Host "======================================="
-		}
-	} catch { Write-Error "Error in writing log: $($_.Exception.Message)" }
+        If ($writeToFile)
+        {
+            $msgToWrite | Out-File -Append -FilePath $LOG_FILE_PATH 
+        }
+        If ($Footer)
+        { 
+            "=======================================" | Out-File -Append -FilePath $LOG_FILE_PATH 
+            Write-Host "======================================="
+        }
+    }
+    catch
+    {
+        Write-Error "Error in writing log: $($_.Exception.Message)" 
+    }
 }
 
-Function Collect-ExceptionMessage
+Function Get-ExceptionMessage
```

Comment:
@bab29 in other scripts I started using Join-ExceptionMessage (to use an acceptable noun)
Would you mind using the Join noun instead of Get? (mainly for alignment of other scripts)?

Reply:
Not a problem, made the update.

Reply:
@AssafMiron  Not a problem. I will note it for other scripts. I made the update.